app/src/main/java/org/oppia/android/app/settings/profile/SnackbarManager.kt

```diff
@@ -0,0 +1,64 @@
+package org.oppia.android.app.settings.profile
+
+import android.util.Log
+import android.view.View
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import javax.inject.Inject
+
+class SnackbarManager @Inject constructor(private val snackbarController: SnackbarController) {
+
+  fun showSnackbar(@StringRes messageStringId: Int, duration: SnackbarController.SnackbarDuration) {
+    snackbarController.enqueueSnackbar(
+      SnackbarController.SnackbarRequest.ShowSnackbar(
+        messageStringId,
+        duration
+      )
+    )
+  }
+
+  fun enableShowingSnackbars(activity: AppCompatActivity) {
+    snackbarController.getCurrentSnackbar().toLiveData().observe(activity) { result ->
+      when (result) {
+        is AsyncResult.Success -> when (val request = result.value) {
+          is SnackbarController.SnackbarRequest.ShowSnackbar -> showSnackbar(
+            activity.findViewById(
+              android.R.id.content
+            ),
+            request
+          )
+          SnackbarController.SnackbarRequest.ShowNothing -> {}
```

Comment:
Doesn't this require dismissing logic for the currently showing snackbar, too?

Also, I think we need to add a callback to the current active snackbar has been dismissed (e.g. via https://developer.android.com/reference/com/google/android/material/snackbar/BaseTransientBottomBar.BaseCallback#onDismissed(B,%20int)) so that we can notify SnackbarController that the snackbar has finished and another may show after it.

Reply:
added this https://github.com/oppia/oppia-android/blob/39740b8d4d798f42e424f17e2fa773ffd86ba5a4/app/src/main/java/org/oppia/android/app/settings/profile/SnackbarManager.kt#L37